### PR TITLE
 StepFormButtonbar and StepFormHeader cleaning

### PR DESCRIPTION
**Pull request by rhuille**
StepFormButtonbar and StepFormHeader cleaning

- remove "cancel" property from 'StepFormButtonbar': cancelEdition is already handled by the 'StepFormHeader'
- remove errors from dom of 'StepFormButtonbar': errors are already displayed inside widgets
- call the 'parent.cancelEdition' method at back click in StepFormHeader: no need to pass a 'cancelEdition' method as prop to StepFormHeader anymore.
- rename 'cancel' event into 'back' event in StepFormHeader
- Use PascalCase for StepFormHeader and StepFormButtonbar in templates

### CODE REVIEW

**Comment by arimbr** [comment]
# [Codecov](https://codecov.io/gh/ToucanToco/weaverbird/pull/433?src=pr&el=h1) Report
> Merging [#433](https://codecov.io/gh/ToucanToco/weaverbird/pull/433?src=pr&el=desc) into [master](https://codecov.io/gh/ToucanToco/weaverbird/commit/19ef23718d936b68bfad569d6abd1bdde02c4ee4&el=desc) will **not change** coverage by `%`.
> The diff coverage is `n/a`.

[Impacted file tree graph](https://codecov.io/gh/ToucanToco/weaverbird/pull/433?src=pr&el=tree)

```diff
@@           Coverage Diff           @@
##           master     #433   +/-   ##
=======================================
  Coverage   96.36%   96.36%           
=======================================
  Files         115      115           
  Lines        2529     2529           
  Branches      363      363           
=======================================
  Hits         2437     2437           
  Misses         92       92           
```



------

[Continue to review full report at Codecov](https://codecov.io/gh/ToucanToco/weaverbird/pull/433?src=pr&el=continue).
> **Legend** - [Click here to learn more](https://docs.codecov.io/docs/codecov-delta)
> `Δ = absolute <relative> (impact)`, `ø = not affected`, `? = missing data`
> Powered by [Codecov](https://codecov.io/gh/ToucanToco/weaverbird/pull/433?src=pr&el=footer). Last update [19ef237...5a219d2](https://codecov.io/gh/ToucanToco/weaverbird/pull/433?src=pr&el=lastupdated). Read the [comment docs](https://docs.codecov.io/docs/pull-request-comments).

**Comment by arimbr** [comment]
# [Codecov](https://codecov.io/gh/ToucanToco/weaverbird/pull/433?src=pr&el=h1) Report
> Merging [#433](https://codecov.io/gh/ToucanToco/weaverbird/pull/433?src=pr&el=desc) into [master](https://codecov.io/gh/ToucanToco/weaverbird/commit/8b6505d3bbbc2e20e5c72465c866ac361cf80650&el=desc) will **not change** coverage by `%`.
> The diff coverage is `n/a`.

[Impacted file tree graph](https://codecov.io/gh/ToucanToco/weaverbird/pull/433?src=pr&el=tree)

```diff
@@           Coverage Diff           @@
##           master     #433   +/-   ##
=======================================
  Coverage   96.36%   96.36%           
=======================================
  Files         115      115           
  Lines        2529     2529           
  Branches      363      363           
=======================================
  Hits         2437     2437           
  Misses         92       92           
```



------

[Continue to review full report at Codecov](https://codecov.io/gh/ToucanToco/weaverbird/pull/433?src=pr&el=continue).
> **Legend** - [Click here to learn more](https://docs.codecov.io/docs/codecov-delta)
> `Δ = absolute <relative> (impact)`, `ø = not affected`, `? = missing data`
> Powered by [Codecov](https://codecov.io/gh/ToucanToco/weaverbird/pull/433?src=pr&el=footer). Last update [8b6505d...d70327c](https://codecov.io/gh/ToucanToco/weaverbird/pull/433?src=pr&el=lastupdated). Read the [comment docs](https://docs.codecov.io/docs/pull-request-comments).

File: tests/unit/utils.ts [hunk 1]
```diff
@@ -174,6 +174,19 @@ export class BasicStepFormTestRunner {
...
+      const cancelEditionCustomMock = jest.fn();
+      const methods = { cancelEdition: cancelEditionCustomMock };
+      const wrapper = mount(this.componentType, {
+        store,
+        localVue: this.vue,
+        sync: false,
+        methods,
+      });
+      wrapper.find('.step-edit-form__back-button').trigger('click');
+      expect(cancelEditionCustomMock.mock.calls.length).toBe(1);
```
Comment (adimascio) [inline tests/unit/utils.ts]:
```suggestion
 expect(cancelEditionCustomMock).toHaveBeenCalledTimes(1);
```

File: docs/_docs/tech/stepforms.md [hunk 2]
```diff
@@ -13,7 +13,7 @@ you can leverage a few components and base classes provided:
   generates a standard step form header bar. It should be fed with a `cancel` event handler and a `title` prop, e.g.:
 
   ```typescript
-  <step-form-header :cancel= "cancelEventHandler" :title="title"></step-form-header>
+  <StepFormHeader :cancel= "cancelEventHandler" :title="title"></StepFormHeader>
```
Comment (adimascio) [inline docs/_docs/tech/stepforms.md]: Shouldn't this part of the doc should be updated with the new `@back` event?

**Pull request merged**